File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_024.cpp
TEST_F(FillFormatIntegrationTest, SolidFill) {
    Presentation pres;
    auto& slide = clear_slide(pres);
    auto& shape = slide.shapes().add_auto_shape(ShapeType::RECTANGLE, 50, 50, 200, 100);
    shape.fill_format().set_fill_type(FillType::SOLID);
    shape.fill_format().solid_fill_color().set_color(
        Color::from_argb(255, 0, 128, 255));

    auto pres2 = save_and_reopen(pres);
    auto& ff = pres2.slides()[0].shapes()[0].fill_format();
    EXPECT_EQ(ff.fill_type(), FillType::SOLID);
    auto c = ff.solid_fill_color().color();
    EXPECT_EQ(c.r(), 0);
    EXPECT_EQ(c.g(), 128);
    EXPECT_EQ(c.b(), 255);
}

File: data/imported/knowledge/slides/python/merged/snippets/snippet_024.py
# Adapted from aspose.org: knowledge/slides/python/merged/snippets/snippet_024.py @ 7f72da4e1423546104b40fa8cebf5b9ae3ce9c91
# Imported under the authorization recorded in
# plans/investigations/evidence/imported-corpus-v1/licensing-resolution-state.md

def test_interactive_sequence_persists(self, tmp_pptx):

        pres = Presentation()

        slide = pres.slides[0]

        trigger = slide.shapes.add_auto_shape(ShapeType.BEVEL, 10, 10, 20, 20)

        target = slide.shapes.add_auto_shape(ShapeType.RECTANGLE, 100, 100, 200, 100)



        seq = slide.timeline.interactive_sequences.add(trigger)

        seq.add_effect(target, anim.EffectType.FADE,

                       anim.EffectSubtype.NONE,

                       anim.EffectTriggerType.ON_CLICK)



        pres2 = tmp_pptx(pres)

        iseqs = pres2.slides[0].timeline.interactive_sequences

        assert len(iseqs) == 1

        assert iseqs[0].count == 1

        assert iseqs[0][0].type == anim.EffectType.FADE

        pres2.dispose()